provider/azure/environ_network.go

```diff
@@ -100,6 +100,33 @@ func (env *azureEnviron) allSubnets(ctx context.ProviderCallContext) ([]network.
 	return results, nil
 }
 
+func (env *azureEnviron) allPublicIPs(ctx context.ProviderCallContext) (map[string]network.ProviderAddress, error) {
+	ipClient := azurenetwork.PublicIPAddressesClient{BaseClient: env.network}
+	ipList, err := ipClient.List(stdcontext.Background(), env.resourceGroup)
```

Comment:
ctx is a context.Context so the ProviderCallContext can be used directly in any SDK API that requires a context

Reply:
Ah! Totally forgot about that. Fixed via https://github.com/juju/juju/pull/13658/commits/a3b8df2b3a0455b4ad13f24eb525e445991fcbd0